### PR TITLE
Spotify oauth

### PR DESCRIPTION
**Pull request by PaulDebevec**
- Add Omniauth and Spotify omniauth initializer
- Add Spotify OAuth
- Create SessionsController to log in 
- Add helper method to ApplicationController for current_user
- Add token_expiration and uid to user table in DB
- Add temporary dashboard 

`rspec`
```Finished in 0.93063 seconds (files took 4.98 seconds to load)
8 examples, 0 failures, 1 pending

Coverage report generated for RSpec to /Users/pauldebevec/turing/module_3/projects/group_project/Jam_Session/coverage. 49 / 63 LOC (77.78%) covered.```

### CODE REVIEW

File: app/controllers/dashboard_controller.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class DashboardController < ApplicationController
+  def index
+
```
Comment (coloniusrex) [inline app/controllers/dashboard_controller.rb:3]: Thanks for the setup!

File: app/controllers/dashboard_controller.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+  end
```
Comment (KirbyDD) [inline app/controllers/dashboard_controller.rb:4]: to save 2 lines you can refactor this to def index; end

File: app/views/sessions/new.html.erb [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+            <%= f.submit 'Log In' %>
+      <% end %>
+    </section>
+  </div>
+</main>
+
+
+
+  <%= button_to "Register", "/register", class: "welcome-btn" %>
```
Comment (Maxwell-Baird) [inline app/views/sessions/new.html.erb:20]: Do we have the link to the spotify login live here?

File: spec/rails_helper.rb [hunk 4]
```diff
@@ -75,9 +82,10 @@
...
-  config.integrate do |with|
-    with.test_framework :rspec
-    with.library :rails
-  end
+
+VCR.configure do |config|
+  config.cassette_library_dir = 'spec/vcr_cassettes'
+  config.hook_into :webmock
+  config.filter_sensitive_data('<SPOTIFY-CLIENT-TOKEN-1>') { ENV['SPOTIFY_CLIENT_ID_1'] }
+  config.configure_rspec_metadata!
```
Comment (Maxwell-Baird) [inline spec/rails_helper.rb:90]: Nice job setting up the VCR

File: app/controllers/dashboard_controller.rb (hunk 2 above)
Comment (PaulDebevec) [reply app/controllers/dashboard_controller.rb:4]: I can do that in this PR - I just know this whole controller is going to be overridden in an upcoming PR when we build out the dashboard.
Thoughts?

File: app/views/sessions/new.html.erb (hunk 3 above)
Comment (PaulDebevec) [reply app/views/sessions/new.html.erb:20]: This was removed - I didn't catch it when I did this commit.

File: app/views/layouts/application.html.erb [hunk 5]
```diff
@@ -22,6 +22,15 @@
       </div>
     </nav>
     <%= image_tag("https://www.whenthereshelpthereshope.com/wp-content/uploads/52265235_l.jpg", class: "welcome-bg") %>
+    <% flash.each do |name, msg| %>
+      <% if name == 'success' %>
```
Comment (coloniusrex) [inline app/views/layouts/application.html.erb:26]: Nice job on this

File: app/views/sessions/new.html.erb (hunk 3 above)
Comment (Maxwell-Baird) [reply app/views/sessions/new.html.erb:20]: Ok just want to check.

File: app/controllers/dashboard_controller.rb (hunk 1 above)
Comment (PaulDebevec) [reply app/controllers/dashboard_controller.rb:3]: No problem! Can't wait to see this built out

**Pull request merged**